### DISCUSSION TITLE
likes, rates and followers with supabase

### DISCUSSION THREAD

**Discussion by marcops85** [supabase/supabase#30431] [Questions]
Hi guys, I have a big dilemma on how to set up the supabase tables for likes, rates and followers. I'll give you an example of how I solved it for now: in the "objects" table I have an id column, one with the object name, one with the description, one with the path image, one with the user id of rate(list), and one with the rates (list).
I like this setting a lot and it works very well because I have everything I need in a single call, especially because in the list I get I can also get the average rating with 5 stars. The problem with this setting is that I have to leave the update RLS open for all authenticated users, and this is a big risk, and problem number 2 in the arrays I can't keep track of the user ids in the list and if a user deletes the 'user id and his vote will remain there forever. 
Many recommend creating a table just for rates with references to the user, the object and obviously the rating. This solves the 2 problems above however so I have the problem that in a listview I can't have both the list of objects and the average of the votes of that object in a single list, I have many more calls to the database, and let's not forget that supabase it doesn't return more than 1000 objects in a single call, so if an object has 2000 rates only half will be returned to me and I will have a wrong average. Ditto for likes and followers. Can anyone give me some advice on what to do?

**Comment by GaryAustin1** [comment]
The 1000 limit is only if you return the data thru the REST API.  You could use a view to do your averaging for you in SQL.  You might event be able to use https://supabase.com/blog/postgrest-aggregate-functions but I'm just skimming thru your details.
